### PR TITLE
Add enum value for `terminationReason` attribute in the guideline document

### LINKED ISSUE #353: An enumeration of `terminationReason` is not provided in API Design Guidelines
**Problem description**
 An enumeration of `terminationReason` is provided in event-subscription-template.yaml

**Expected behavior**
In order to keep API Design Guidelines as normative and artifacts as informative the enumeration of `terminationReason` should be added to  API Design Guidelines https://github.com/camaraproject/Commonalities/blob/main/documentation/API-design-guidelines.md#subscription-ends-event

**Alternative solution**

**Additional context**

### PR DESCRIPTION
**Pull request by bigludo7**
#### What type of PR is this?

* documentation

#### What this PR does / why we need it:

Add enum value for `terminationReason` attribut by adding a table in our documentation.

#### Which issue(s) this PR fixes:

Fixes #353 

#### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

#### Changelog input

```
 release-note

```

#### Additional documentation 

This section can be blank.

```
docs

```

### CODE REVIEW

File: documentation/API-design-guidelines.md [hunk 1]
```diff
@@ -1956,9 +1956,21 @@ Note: For operational and troubleshooting purposes it is relevant to accommodate
...
+It is used when the `subscriptionExpireTime` or `subscriptionMaxEvents` has been reached, or, if the API server has to stop sending notifications prematurely, or if the subscription request is managed asynchronously by the server and it is not able to provide the service. For this specific event, the `data` must feature `terminationReason` attribute. 
 
-Note: The "subscription-ends" notification is not counted in the `subscriptionMaxEvents`. (for example, if a client request a `subscriptionMaxEvents` to 2, later, received second notification, then a third notification will be sent for "subscription-ends")
+Following table list value for `terminationReason` attribut:
```
Comment (rartych) [inline documentation/API-design-guidelines.md:1961]: Some typos:
```suggestion
The following table lists values for `terminationReason` attribute:
```

File: documentation/API-design-guidelines.md [hunk 2]
```diff
@@ -1956,9 +1956,21 @@ Note: For operational and troubleshooting purposes it is relevant to accommodate
...
+It is used when the `subscriptionExpireTime` or `subscriptionMaxEvents` has been reached, or, if the API server has to stop sending notifications prematurely, or if the subscription request is managed asynchronously by the server and it is not able to provide the service. For this specific event, the `data` must feature `terminationReason` attribute. 
 
-Note: The "subscription-ends" notification is not counted in the `subscriptionMaxEvents`. (for example, if a client request a `subscriptionMaxEvents` to 2, later, received second notification, then a third notification will be sent for "subscription-ends")
+The following table lists values for `terminationReason` attribute:
+
+| enum value | termination reason |
+| -----------|--------------------|
+| NETWORK_TERMINATED | API server stopped sending notification|
```
Comment (PedroDiez) [inline documentation/API-design-guidelines.md:1965]:
```suggestion
| NETWORK_TERMINATED | API server stopped sending notification |
```

File: documentation/API-design-guidelines.md [hunk 3]
```diff
@@ -1956,9 +1956,21 @@ Note: For operational and troubleshooting purposes it is relevant to accommodate
...
+| SUBSCRIPTION_EXPIRED | Subscription expire time (optionally set by the requester) has been reached|
+| MAX_EVENTS_REACHED | Maximum number of events (optionally set by the requester) has been reached|
+| ACCESS_TOKEN_EXPIRED | Access Token sinkCredential (optionally set by the requester) expiration time has been reached|
+| SUBSCRIPTION_DELETED | Subscription was deleted by the requester|
+
+Note1:This enumeration is also defined in event-subscription-template.yaml (see in [Commonalities/artifacts/camara-cloudevents](/artifacts/camara-cloudevents) directory ``event-subscription-template.yaml``).
```
Comment (PedroDiez) [inline documentation/API-design-guidelines.md:1971]:
```suggestion
Note1: This enumeration is also defined in `event-subscription-template.yaml` (placed in [Commonalities/artifacts/camara-cloudevents](/artifacts/camara-cloudevents) directory). 
```

File: documentation/API-design-guidelines.md (hunk 2 above)
Comment (PedroDiez) [reply documentation/API-design-guidelines.md:1965]: adding a space in every row

File: documentation/API-design-guidelines.md (hunk 3 above)
Comment (PedroDiez) [reply documentation/API-design-guidelines.md:1971]: Some suggestion at your discretion

**Review by PedroDiez — COMMENTED** [review]
Some editorial comments

**Comment by bigludo7** [comment]
> Some editorial comments

All took in to consideration - Thanks

**Review by rartych — DISMISSED** [review]
LGTM

File: documentation/API-design-guidelines.md [hunk 4]
```diff
@@ -1956,9 +1956,21 @@ Note: For operational and troubleshooting purposes it is relevant to accommodate
...
+It is used when the `subscriptionExpireTime` or `subscriptionMaxEvents` has been reached, or, if the API server has to stop sending notifications prematurely, or if the subscription request is managed asynchronously by the server and it is not able to provide the service. For this specific event, the `data` must feature `terminationReason` attribute. 
 
-Note: The "subscription-ends" notification is not counted in the `subscriptionMaxEvents`. (for example, if a client request a `subscriptionMaxEvents` to 2, later, received second notification, then a third notification will be sent for "subscription-ends")
+The following table lists values for `terminationReason` attribute:
+
+| enum value | termination reason |
+| -----------|--------------------|
```
Comment (PedroDiez) [inline documentation/API-design-guidelines.md:1964]:
```suggestion
| -----------|-------------------- |
```

File: documentation/API-design-guidelines.md [hunk 5]
```diff
@@ -1956,9 +1956,21 @@ Note: For operational and troubleshooting purposes it is relevant to accommodate
...
+| NETWORK_TERMINATED | API server stopped sending notification |
+| SUBSCRIPTION_EXPIRED | Subscription expire time (optionally set by the requester) has been reached|
```
Comment (PedroDiez) [inline documentation/API-design-guidelines.md:1966]:
```suggestion
| SUBSCRIPTION_EXPIRED | Subscription expire time (optionally set by the requester) has been reached |
```

File: documentation/API-design-guidelines.md [hunk 6]
```diff
@@ -1956,9 +1956,21 @@ Note: For operational and troubleshooting purposes it is relevant to accommodate
...
+| MAX_EVENTS_REACHED | Maximum number of events (optionally set by the requester) has been reached|
```
Comment (PedroDiez) [inline documentation/API-design-guidelines.md:1967]:
```suggestion
| MAX_EVENTS_REACHED | Maximum number of events (optionally set by the requester) has been reached |
```

File: documentation/API-design-guidelines.md [hunk 7]
```diff
@@ -1956,9 +1956,21 @@ Note: For operational and troubleshooting purposes it is relevant to accommodate
...
+| ACCESS_TOKEN_EXPIRED | Access Token sinkCredential (optionally set by the requester) expiration time has been reached|
```
Comment (PedroDiez) [inline documentation/API-design-guidelines.md:1968]:
```suggestion
| ACCESS_TOKEN_EXPIRED | Access Token sinkCredential (optionally set by the requester) expiration time has been reached |
```

File: documentation/API-design-guidelines.md [hunk 8]
```diff
@@ -1956,9 +1956,21 @@ Note: For operational and troubleshooting purposes it is relevant to accommodate
...
+| SUBSCRIPTION_DELETED | Subscription was deleted by the requester|
```
Comment (PedroDiez) [inline documentation/API-design-guidelines.md:1969]:
```suggestion
| SUBSCRIPTION_DELETED | Subscription was deleted by the requester |
```

**Comment by PedroDiez** [comment]
Just committed som editorials @bigludo7, @rartych

**Review by PedroDiez — APPROVED** [review]
LGTM

**Review by rartych — APPROVED** [review]

**Pull request merged**